packages/edit-post/src/components/start-page-options/index.js

```diff
@@ -19,11 +19,20 @@ import { store as editPostStore } from '../../store';
 
 function PatternSelection( { onChoosePattern } ) {
 	const { blockPatterns } = useSelect( ( select ) => {
-		const { __experimentalGetPatternsByBlockTypes } =
-			select( blockEditorStore );
+		const { __experimentalGetPatternsByBlockTypes } = select(
+			blockEditorStore
+		);
+		const { getCurrentPostType } = select( editorStore );
+		const postType = getCurrentPostType();
 		return {
-			blockPatterns:
-				__experimentalGetPatternsByBlockTypes( 'core/post-content' ),
+			blockPatterns: ( postType === 'page'
+				? __experimentalGetPatternsByBlockTypes( 'core/post-content' )
+				: []
+			).concat(
+				__experimentalGetPatternsByBlockTypes(
+					`core/post-content/${ postType }`
```

Comment:
This is a very weird construct to me since it seems to imply the post type is a child of the block type. Why can't we go with a separate key for post types? Seems both clearer and more flexible.